cmake/all.cmake

```diff
@@ -162,6 +166,12 @@ set(CATKIN_GLOBAL_LIBEXEC_DESTINATION lib)
 set(CATKIN_GLOBAL_PYTHON_DESTINATION ${PYTHON_INSTALL_DIR})
 set(CATKIN_GLOBAL_SHARE_DESTINATION share)
 
+# There is an issue with bootstrapping catkin_make_isolated on Windows.  Without a toplevel
+# workspace to call this function, several shared libraries are built incorrectly.
+if (MSVC)
+  configure_shared_library_build_settings()
+endif()
```

Comment:
Currently `configure_shared_library_build_settings()` is being called only conditionally on Windows. That seems wrong to me. Shouldn't this be done for all platforms?

Anyway this change has the potential of breaking existing use cases.